core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/configuration/ConfigurationLoaderImpl.java

```diff
@@ -55,32 +57,63 @@
 	private CommandLineArgumentHandler cmaHandler;
 
 	private static final String DEFAULT_CONFIG_DIR = "config";
-	private Path configDir;
+	private List<Path> configDirs;
 
 	@PostConstruct
 	private void setup() {
-		String configDirS = DEFAULT_CONFIG_DIR;
+		this.configDirs = new ArrayList<>();
+		this.configDirs.add(this.environment.getWorkingDirectory().resolve(DEFAULT_CONFIG_DIR));
 
-		String cmaConfigPath = this.cmaHandler.getConfigPath();
-		if (cmaConfigPath != null) {
-			configDirS = cmaConfigPath;
+		List<String> cmaConfigPaths = this.cmaHandler.getConfigPaths();
+		if (cmaConfigPaths != null) {
+			for (String path : cmaConfigPaths) {
+				this.configDirs.add(this.environment.getWorkingDirectory().resolve(path));
+			}
 		}
 
-		this.configDir = this.environment.getWorkingDirectory().resolve(configDirS);
-		if (!this.configDir.toFile().isDirectory()) {
-			this.logger.error("the configuration directory {} does not exists", this.configDir.toAbsolutePath());
+		boolean found = false;
+		for (Path p : this.configDirs) {
+			if (p.toFile().isDirectory()) {
+				found = true;
+			} else {
+				this.logger.warn("Configuration directory {} does not exist", p);
+			}
+		}
+		if (!found) {
+			this.logger.error("No valid configuration directory found.");
 		}
 	}
 
 	/**
+	 * Finds the configuration file for the given name in all config directorys. The config dirs are checked from last
+	 * to first.
 	 * 
 	 * @param configurationName
-	 *            the name of the config file, without the .properties
-	 * @return the loaded Properties
+	 *            the name of the configuration to search
+	 * @return The first path that matches the name or null if no such file was found.
+	 */
+	private Path findPropertiesFile(String configurationName) {
+		for (int i = this.configDirs.size() - 1; i >= 0; i--) {
+			Path p = this.configDirs.get(i).resolve(configurationName + ".properties");
+			if (p.toFile().exists())
+				return p;
+		}
+		return null;
+	}
+
+	/**
+	 * @see de.unistuttgart.iaas.amyassist.amy.core.configuration.ConfigurationLoader#load(java.lang.String)
 	 */
+	@Override
 	public Properties load(String configurationName) {
 		Properties properties = new Properties();
-		Path path = this.configDir.resolve(configurationName + ".properties");
+		Path path = findPropertiesFile(configurationName);
```

Comment:
`this.`

Reply:
Yes. Actually don't know how this happened.